### DISCUSSION TITLE
How can we import python files?

### DISCUSSION THREAD

**Discussion by CaiYitao** [cjdoris/PythonCall.jl#145] [Q&A] [answered]
How can we import python files(like the somepyfile.py from github or local python files)?
```julia
pyimport("my_python_file.py")
```
this seems not enough.

**Comment by cjdoris** [comment, marked as answer]
In Python, do don't import files, you import packages. Either:
- Turn your file into a package and import that.
- If you just want to execute the contents of the file, you can do something like `pyexec(read("my_python_file.py"))`.

**Reply by CaiYitao** [reply]
Do you mean turning the github repo into a pip/conda installable package first? and then import?

Is there easier way to import a github repo?  for example like [this repo ](https://github.com/octavian-ganea/equidock_public)  it is not an installable package, just a code companion for paper.

**Reply by cjdoris** [reply]
That repo consists of a bunch of scripts. You don't import scripts, you execute them. In Julia you execute something like ``run(`python some/script.py`)``.

You need to figure out how to do what you want to do using Python first before using PythonCall to do it.

**Reply by CaiYitao** [reply]
Normally I would edit some classes or functions to my needs, and mess around with the models in a notebook,   not just simply run those scripts.
